zerver/lib/event_queue.py

```diff
@@ -94,10 +98,12 @@ def to_dict(self):
                     client_type_name=self.client_type_name)
 
     def __repr__(self):
+        # type: () -> str
         return "ClientDescriptor<%s>" % (self.event_queue.id,)
 
     @classmethod
     def from_dict(cls, d):
+        # type: (Dict[str, Any]) -> ClientDescriptor
```

Comment:
I think `d` should be typed as `MutableMapping[str, Any]` instead. It is better to use duck types in parameters. See [cheat sheet](https://github.com/python/mypy/blob/master/docs/source/cheat_sheet.rst#standard-duck-types).